controlplane/pkg/tests/request_builder_test.go

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2019 Cisco and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/connection"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/connection/mechanisms/kernel"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/connection/mechanisms/vxlan"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/networkservice"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/registry"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/common"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/model"
+)
+
+type testModel struct {
+	model.Model
+}
+
+func (m *testModel) ConnectionID() string {
+	return "1"
+}
+
+func newTestModel() *testModel {
+	m := &testModel{
+		Model: model.NewModel(),
+	}
+	testNsm := &registry.NetworkServiceManager{
+		Name: "nsm1",
+	}
+	m.SetNsm(testNsm)
+	return m
+}
+
+func getTestFwdWithLocalMechanism() *model.Forwarder {
+	return &model.Forwarder{
+		LocalMechanisms: []*connection.Mechanism{
+			{
+				Type: kernel.MECHANISM,
+			},
+		},
+	}
+}
+
+func getTestFwdWithRemoteMechanism() *model.Forwarder {
+	return &model.Forwarder{
+		RemoteMechanisms: []*connection.Mechanism{
+			{
+				Type: vxlan.MECHANISM,
+			},
+		},
+	}
+}
+
+func getLocalTestEndpoint() *registry.NSERegistration {
```

Comment:
'get' means that object already exists and the function 'fetches' it. However, a new structure is generated for every call thus it seems more accurate to name the function {new|create}LocalTestEndpoint.